File: app/src/main/java/cn/com/example/wendi/rollingball/Option.java
package cn.com.example.wendi.rollingball;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.content.pm.ActivityInfo;
import android.graphics.Typeface;
import android.os.Bundle;
import android.os.Vibrator;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;


public class Option extends Activity {

    /**
     * onCreate method
     * set the buttons
     * @param savedInstanceState
     */

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_option);
        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);

        SharedPreferences sharedPref = Option.this.getSharedPreferences("Scores",MODE_WORLD_READABLE);
        long highScore = sharedPref.getInt(getString(R.string.saved_high_score), 0);
        TextView tv = (TextView) findViewById(R.id.high_S);
        tv.setText(String.valueOf(highScore));
        Typeface face = Typeface.createFromAsset (  getAssets() , "fonts/Chalkduster.ttf" );
        tv.setTypeface(face);

        Button bk = (Button)findViewById(R.id.bck);
        bk.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View arg){
                Vibrator vb = (Vibrator)getSystemService(Context.VIBRATOR_SERVICE);
                long [] pattern = {0,20};
                vb.vibrate(pattern,-1);
                Option.this.finish();
            }
        });

        /**
         * user can clear high score record here
         */
        Button cr = (Button) findViewById(R.id.del);
        cr.setOnClickListener(new View.OnClickListener(){
            @Override
        public void onClick(View arg){
                Vibrator vb = (Vibrator)getSystemService(Context.VIBRATOR_SERVICE);
                long [] pattern = {0,20};
                vb.vibrate(pattern,-1);
                SharedPreferences sharedPref = Option.this.getSharedPreferences("Scores", MODE_WORLD_READABLE);
                SharedPreferences.Editor editor = sharedPref.edit();
                editor.putInt(getString(R.string.saved_high_score), 0);
                editor.commit();
                TextView tv = (TextView) findViewById(R.id.high_S);
                tv.setText("0");
                Typeface face = Typeface.createFromAsset (  getAssets() , "fonts/Chalkduster.ttf" );
                tv.setTypeface(face);
            }
        });
    }


}

File: app/src/main/kotlin/cn/com/example/wendi/rollingball/Option.kt
package cn.com.example.wendi.rollingball

import android.content.Context
import android.content.SharedPreferences
import android.content.pm.ActivityInfo
import android.graphics.Typeface
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.edit

class Option : AppCompatActivity() {

    /**
     * onCreate method
     * set the buttons
     * @param savedInstanceState
     */
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_option)
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE

        val sharedPref = getSharedPreferences("Scores", Context.MODE_PRIVATE)
        val highScore = sharedPref.getInt(getString(R.string.saved_high_score), 0).toLong()
        val tv = findViewById<TextView>(R.id.high_S)
        tv.text = highScore.toString()
        val face = Typeface.createFromAsset(assets, "fonts/Chalkduster.ttf")
        tv.typeface = face

        val bk = findViewById<Button>(R.id.bck)
        bk.setOnClickListener {
            vibrateInGame(VibrationType.CLICK)
            finish()
        }

        /**
         * user can clear high score record here
         */
        val cr = findViewById<Button>(R.id.del)
        cr.setOnClickListener {
            vibrateInGame(VibrationType.CLICK)
            sharedPref.edit {
                putInt(getString(R.string.saved_high_score), 0)
            }
            tv.text = "0"
            tv.typeface = face
        }
    }
}
